tasks/sep/032-op-pectra-defence/VALIDATION.md

```diff
@@ -0,0 +1,126 @@
+
+## State Changes
+
+Note: The changes listed below do not include safe nonce updates or liveness guard related changes. Refer to the
+[generic nested Safe execution validation document](../../../NESTED-VALIDATION.md)
+
+### `0x05F9613aDB30026FFd634f38e5C4dFd30a197Fa1` (`DisputeGameFactoryProxy`)
+
+Click on 'show raw state changes'.
+
+- **Key**: `0x4d5a9bd2e41301728d41c8e705190becb4e74abe869f75bdb405b63716a35f9e` <br/>
+  **Before**: `0x000000000000000000000000bbdbdfe37c02439764de0e41c906e4396b5b3914` <br/>
+  **After**: `0x0000000000000000000000006b8a1c71e23e103d3a81e7878656be9516733ca3` <br/>
+  **Meaning**: Updates the implementation for game type 1. Verify that the old implementation is set in this slot using
+  `cast call 0x05F9613aDB30026FFd634f38e5C4dFd30a197Fa1 "gameImpls(uint32)(address)" 1`.
+
+- **Key**: `0xffdfc1249c027f9191656349feb0761381bb32c9f557e01f419fd08754bf5a1b` <br/>
+  **Before**: `0x000000000000000000000000f3ccf0c4b51d42cfe6073f0278c19a8d1900e856` <br/>
+  **After**: `0x000000000000000000000000927248cb1255e0f02413a758899db4aecffaa5fe` <br/>
+  **Meaning**: Updates the implementation for game type 0. Verify that the old implementation is set using
+  `cast call 0x05F9613aDB30026FFd634f38e5C4dFd30a197Fa1 "gameImpls(uint32)(address)" 0`.
+
+
+## Verifying Dispute Games
+
+<!-- Sepolia todo: We should convert this to solidity and run the checks in the postCheck method -->
+The old and new dispute game contracts can be compared with the [comparegames.sh](https://gist.github.com/ajsutton/28be852a36d9d19af16f7c870b267873)
+script.
+
+From the tenderly simulation click on 'Run on Fork', then copy and export the RPC url provided.
+
+```
+export ETH_RPC_URL=https://rpc.tenderly.co/fork/...
+```
+
+The arguments to the script can be taken from the before and after values in the `DisputeGameFactoryProxy` above.
+
+### PermissionedDisputeGame:
+
+The only change seen here is the `absolutePrestate()` as expected.
+
+```shell
+comparegames.sh 0xbbdbdfe37c02439764de0e41c906e4396b5b3914 0x6b8a1c71e23e103d3a81e7878656be9516733ca3
+
+Mismatch version()(string)
+Was: "1.3.1"
+Now: "1.3.1-beta.3"
+
+Mismatch absolutePrestate()(bytes32)
+Was: 0x03f89406817db1ed7fd8b31e13300444652cdb0b9c509a674de43483b2f83568
+Now: 0x035ac388b5cb22acf52a2063cfde108d09b1888655d21f02f595f9c3ea6cbdcd
+
+Matches maxGameDepth()(uint256) = 73
+
+Matches splitDepth()(uint256) = 30
+
+Matches maxClockDuration()(uint256) = 302400 [3.024e5]
+
+Matches gameType()(uint32) = 1
+
+Matches l2ChainId()(uint256) = 11155420 [1.115e7]
+
+Matches clockExtension()(uint64) = 10800 [1.08e4]
+
+Matches anchorStateRegistry()(address) = 0x218CD9489199F321E1177b56385d333c5B598629
+Matches weth()(address)
+Matches vmAn()(address)
+```
+
+### FaultDisputeGame:
+
+There are two changes here:
+1. the `absolutePrestate()` as expected.
+2. The version. This is because the FDG was manually added to the balrog devnet from `develop`.
```

Comment:
The version matches for the FDG, I think you meant to put this in the PDG section and update the text?

Reply:
this is not longer relevant. The new OPPU deploys matching versions.